experimental/dds/tree2/src/core/tree/cursor.ts

```diff
@@ -363,6 +363,36 @@ export function forEachNode<TCursor extends ITreeCursor = ITreeCursor>(
 	}
 }
 
+/**
+ * Moves cursor to each child of the current field.  The optional arguments `start` and `end`
+ * can be used to restrict the range of children visited.
+ *
+ * `callbackFn` is invoked after moving the cursor to each child.  The callback must return
+ * `true` to continue the enumeration, or `false` to early exit.
+ *
+ * The `callbackFn` may move move the provided cursor, but must restore the cursor to the current
+ * child node in the enumeration before returning.
+ */
+export function forEachNodeIn<TCursor extends ITreeCursor = ITreeCursor>(
```

Comment:
@CraigMacomber - Slightly enhanced version of 'forEachNode' that allows specifying a range to visit (for APIs that support subsets, like .slice()) and allows early exit if the callback returns false (for APIs that early-exit, like indexOf).

Reply:
Pretty much every function we have that puts the cursor is a very clear and predictable position before returning, which is almost always there it started. This appears to sometimes leave it somewhere in the field, and sometimes move it back up to where it started? I think thats a bug, or at least an oversight?